pyfpga/templates/diamond.jinja

```diff
@@ -0,0 +1,135 @@
+{#
+#
+# Copyright (C) 2015-2024 PyFPGA Project
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+#}
+
+{%- if 'cfg' in steps -%}
+
+# ---------------------
+# Project configuration
+# ---------------------
+
+# Project creation
+set impl_name "impl1"
```

Comment:
The reason I used this variable is that this gives the user an easy way to change implementations in a pre/post-run tcl. But I guess since everything is tool-generated anyway, it doesn't really make a difference if we have two implementations and choose which one or change a constant so that pyFPGA modifies impl1 according to the required changes each time.

I'm open to either solution.

Reply:
I would like to maintain pyfpga simple, so I implemented a simple flow, where hooks are available if customizations are needed. Things such as different implementations are not supported by all the tools, not in a homogenous way. I prefer to hard-code the default name impl1.